### PR TITLE
issue-7 Replace while

### PR DESCRIPTION
**Pull request by tr4rex**
To get rid of using while I can imagine only 7 ways - streams or recursive function. So I have chosen streams

### CODE REVIEW

**Comment by jcranky** [comment]
Is there a test for that? Or perhaps this is a good opportunity for adding one?

**Comment by tr4rex** [comment]
good idea

**Comment by jcranky** [comment]
@tr4rex you can also have a look at #14  as a starting point

**Comment by tr4rex** [comment]
at the moment I need specs2 dependency, which is in #14 . so w8ing for that, meanwhile will take another issue

**Comment by jcranky** [comment]
@tr4rex #14 merged

**Comment by tr4rex** [comment]
@jcranky tests added

File: project/Dependencies.scala [hunk 1]
```diff
@@ -7,6 +7,10 @@ object Dependencies {
     "org.specs2" %% "specs2-scalacheck" % specs2Version % Test
   )
 
+  val mockito = Seq(
+    "org.specs2" %% "specs2-mock" % specs2Version % Test
```
Comment (jcranky) [inline project/Dependencies.scala:8]: since this is a specs2 module, perhaps it would be better to put this together with the other specs2 dependencies

**Review by jcranky — COMMENTED** [review]
Nice work with the tests. Did it uncover any bugs?

File: project/Dependencies.scala (hunk 1 above)
Comment (tr4rex) [reply project/Dependencies.scala:8]: done

**Comment by tr4rex** [comment]
nope. i just refactored a little bit the code. but probably there are a lot of mistakes, taking into account how often var and foreach are used

**Review by jcranky — APPROVED** [review]

**Pull request merged**